crates/subgraph/src/performance/order_performance.rs

```diff
@@ -0,0 +1,765 @@
+use super::apy::APYDetails;
+use super::vol::VolumeDetails;
+use super::{PerformanceError, YEAR18};
+use crate::performance::apy::{get_vaults_apy, TokenPair};
+use crate::{
+    performance::vol::get_vaults_vol,
+    types::common::{Erc20, Order, Trade},
+};
+use alloy::primitives::U256;
+use rain_orderbook_math::{BigUintMath, ONE18};
+use serde::{Deserialize, Serialize};
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use typeshare::typeshare;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Hash)]
+#[serde(rename_all = "camelCase")]
+#[typeshare]
+pub struct VaultPerformance {
+    /// vault id
+    pub id: String,
+    /// vault token
+    pub token: Erc20,
+    /// vault vol segment
+    pub vol_details: VolumeDetails,
+    /// vault apy segment
+    pub apy_details: Option<APYDetails>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[typeshare]
+pub struct DenominatedPerformance {
+    /// The denomination token
+    pub token: Erc20,
+    /// Order's APY raw value
+    pub apy: U256,
+    /// Determines if apy is negative or not
+    pub apy_is_neg: bool,
+    /// Order's net vol raw value
+    pub net_vol: U256,
+    /// Determines if net_vol is negative or not
+    pub net_vol_is_neg: bool,
+    /// Order's starting capital
+    pub starting_capital: U256,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[typeshare]
+pub struct OrderPerformance {
+    /// Order subgraph id
+    pub order_id: String,
+    /// Order hash
+    pub order_hash: String,
+    /// Order's orderbook
+    pub orderbook: String,
+    /// Order's measured performance as a whole
+    pub denominated_performance: Option<DenominatedPerformance>,
+    /// Start timestamp of the performance measring timeframe
+    #[typeshare(typescript(type = "number"))]
+    pub start_time: u64,
+    /// End timestamp of the performance measuring timeframe
+    #[typeshare(typescript(type = "number"))]
+    pub end_time: u64,
+    /// Order's input vaults isolated performance
+    pub inputs_vaults: Vec<VaultPerformance>,
+    /// Order's output vaults isolated performance
+    pub outputs_vaults: Vec<VaultPerformance>,
+}
+
+impl OrderPerformance {
+    /// Given an order and its trades and optionally a timeframe, will calculates
+    /// the order performance, (apy and volume)
+    /// Trades must be sorted in desc order by timestamp, this is the case if
+    /// queried from subgraph using this lib functionalities
+    pub fn measure(
```

Comment:
test?

Reply:
`test_get_order_performance()` on this file

Reply:
Ill rename it for clarification